npm/webpack-dev-server/src/devServer.ts

```diff
@@ -115,6 +116,29 @@ export type PresetHandlerResult = { frameworkConfig: Configuration, sourceWebpac
 type Optional<T, K extends keyof T> = Pick<Partial<T>, K> & Omit<T, K>
 
 async function getPreset (devServerConfig: WebpackDevServerConfig): Promise<Optional<PresetHandlerResult, 'frameworkConfig'>> {
+  const defaultWebpackModules = () => ({ sourceWebpackModulesResult: sourceDefaultWebpackDependencies(devServerConfig) })
+
+  // Third party framework
+  if (devServerConfig.framework?.startsWith('cypress-ct-')) {
+    let frameworkConfig: Configuration | undefined
+
+    try {
+      // eg `import('cypress-ct-solid-js')
+      const mod = await importModule(devServerConfig.framework)
```

Comment:
We will probably need something like this in `cypress/vite-dev-server`, too. We don't have that yet.